assert/assertions.go

```diff
@@ -1167,58 +1160,25 @@ func WithinRange(t TestingT, actual, start, end time.Time, msgAndArgs ...interfa
 	return true
 }
 
-func toFloat(x interface{}) (float64, bool) {
-	var xf float64
-	xok := true
-
-	switch xn := x.(type) {
-	case uint:
-		xf = float64(xn)
-	case uint8:
-		xf = float64(xn)
-	case uint16:
-		xf = float64(xn)
-	case uint32:
-		xf = float64(xn)
-	case uint64:
-		xf = float64(xn)
-	case int:
-		xf = float64(xn)
-	case int8:
-		xf = float64(xn)
-	case int16:
-		xf = float64(xn)
-	case int32:
-		xf = float64(xn)
-	case int64:
-		xf = float64(xn)
-	case float32:
-		xf = float64(xn)
-	case float64:
-		xf = xn
-	case time.Duration:
-		xf = float64(xn)
-	default:
-		xok = false
-	}
+type ConvertibleToFloat64 interface {
+	uint | uint8 | uint16 | uint32 | uint64 | int | int8 | int16 | int32 | int64 | float32 | float64 | time.Duration
+}
 
-	return xf, xok
+func toFloat[T ConvertibleToFloat64](x T) float64 {
+	return float64(x)
 }
 
+
 // InDelta asserts that the two numerals are within delta of each other.
 //
-// 	 assert.InDelta(t, math.Pi, 22/7.0, 0.01)
-func InDelta(t TestingT, expected, actual interface{}, delta float64, msgAndArgs ...interface{}) bool {
+//	assert.InDelta(t, math.Pi, 22/7.0, 0.01)
+func InDelta[T ConvertibleToFloat64](t TestingT, expected, actual T, delta float64, msgAndArgs ...interface{}) bool {
```

Comment:
InDelta should only take `~float32 | ~float64`. `ConvertibleToFloat64` is too wide.